src/components/sharing/util.js

```diff
@@ -0,0 +1,431 @@
+        // permission list
+        const users = resp[type].reduce((currUsers, sharing) => {
+            const permList = permListFn(sharing);
+            return [...currUsers, ...permList.map((perm) => userIdFn(perm))];
+        }, []);
+        return [...acc, ...users];
+    }, []);
+    return getUniqueList(allUsers);
+};
+
+// Takes the resources provided from the bag and returns the total #
+export const getResourceTotal = (resources) => {
+    return Object.values(resources).reduce((acc, type) => {
+        return acc + type.length;
+    }, 0);
+};
+
+// Checks a response from the sharing endpoints and will find which resource
+// type key exists and will return the values
+export const getShareResponseValues = (resp) => {
+    return (
+        resp[TYPE.DATA] ||
+        resp[TYPE.APPS] ||
+        resp[TYPE.ANALYSES] ||
+        resp[TYPE.TOOLS]
+    );
+};
+
+/**
+ * Takes the userMap returned from `groupByUser`, and returns all the
+ * sharing requests that need to be sent out for each resource type.
+ * Will return an object that looks like
+ * `{paths: [updates], apps: [updates], analyses: [updates], tools: [updates]}`
+ * @param userMap
+ */
+export const getSharingUpdates = (userMap) => {
+    return Object.keys(userMap).reduce((acc, userId) => {
+        const user = userMap[userId];
+        const { originalPermission, displayPermission } = user;
+
+        if (originalPermission !== displayPermission) {
+            return allResourceTypes.reduce(
+                (acc, type) => {
+                    const resources = user[type];
+
+                    if (resources && resources.length > 0) {
+                        const { fmtSubjectFn } = getSharingFns(type);
+                        const updates = resources.filter(
+                            (resource) =>
+                                displayPermission !== resource.permission
+                        );
+
+                        if (updates && updates.length > 0) {
+                            const sharingReq = {
+                                ...fmtSubjectFn(user),
+                                [type]: updates.map((resource) => {
+                                    return {
+                                        ...resource,
+                                        permission: displayPermission,
+                                    };
+                                }),
+                            };
+                            return {
+                                ...acc,
+                                [type]: [...(acc[type] || []), sharingReq],
+                            };
+                        } else {
+                            return { ...acc };
+                        }
+                    } else {
+                        return { ...acc };
+                    }
+                },
+                { ...acc }
+            );
+        } else {
+            return { ...acc };
+        }
+    }, {});
```

Comment:
Since something is being returned inside these if-checks, I don't think the `else` blocks are necessary:
```javascript
if (check) {
  return ...;
}
return acc;
```
Again, I don't think the `{ ...acc }` copies are required, especially since it's being created by this function in the first place as an initial `{}` in the top-level reducer.